src/v/datalake/tests/proto_to_arrow_test_utils.h

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+#include "model/record.h"
+#include "random/generators.h"
+
+#include <seastar/core/circular_buffer.hh>
+
+#include <compression/compression.h>
+#include <google/protobuf/message.h>
+
+#include <cstdlib>
+#include <sstream>
+#include <string>
+
+namespace datalake {
+// This is here for now to split up the stack of commits into separate PRs
+struct schema_info {
+    std::string schema;
+};
+} // namespace datalake
+
+struct test_data {
+    std::string empty_schema = (R"schema(
+      syntax = "proto2";
+package datalake.proto;
+
+message empty_message {}
+    )schema");
+
+    std::string simple_schema = (R"schema(
+      syntax = "proto2";
+package datalake.proto;
+
+message simple_message {
+  optional string label = 1;
+  optional int32 number = 3;
+  optional int64 big_number = 4;
+  optional float float_number = 5;
+  optional double double_number = 6;
+  optional bool true_or_false = 7;
+}
+    )schema");
+
+    std::string nested_schema = (R"schema(
+syntax = "proto2";
+package datalake.proto;
+
+message inner_message_t {
+  optional string inner_label = 1;
+  optional int32 inner_number = 2;
+}
+
+message nested_message {
+  optional string label = 1;
+  optional int32 number = 2;
+  optional inner_message_t inner_message = 3;
+}
+)schema");
+
+    std::string combined_schema = (R"schema(
+syntax = "proto2";
+package datalake.proto;
+
+message empty_message {}
+
+message simple_message {
+  optional string label = 1;
+  optional int32 number = 3;
+  optional int64 big_number = 4;
+  optional float float_number = 5;
+  optional double double_number = 6;
+  optional bool true_or_false = 7;
+}
+
+message inner_message_t {
+  optional string inner_label = 1;
+  optional int32 inner_number = 2;
+}
+
+message nested_message {
+  optional string label = 1;
+  optional int32 number = 2;
+  optional inner_message_t inner_message = 3;
+}
+)schema");
+
+    std::string test_message_name = "simple_message";
+};
+
+// TODO: Make this a fixture?
+struct test_message_builder {
+    test_message_builder(test_data data)
+      : _proto_input_stream{data.combined_schema.c_str(), int(data.combined_schema.size())}
+      , _tokenizer{&_proto_input_stream, nullptr} {
```

Comment:
Should the tokenizer have an error collector?

Reply:
I don't think we need that for the test. We should have any errors constructing test messages, and if we do, the test will fail.